data-runtime/src/main/java/io/micronaut/data/runtime/operations/internal/sql/AbstractSqlRepositoryOperations.java

```diff
@@ -211,8 +214,12 @@ protected void setStatementParameter(PS preparedStatement, int index, DataType d
                 }
                 break;
             case JSON:
-                if (value != null && jsonCodec != null && !value.getClass().equals(String.class)) {
-                    value = new String(jsonCodec.encode(value), StandardCharsets.UTF_8);
+                if (value != null && objectMapper  != null && !value.getClass().equals(String.class)) {
+                    try {
+                        value = new String(objectMapper.writeValueAsBytes(value), StandardCharsets.UTF_8);
```

Comment:
Throw an exception if the object mapper is null with some message that it needs to be added

Reply:
Changed